lib/src/http/common/request.dart

```diff
@@ -415,16 +446,23 @@ abstract class CommonRequest extends Object
           !(response.status >= 200 && response.status < 300)) {
         throw new RequestException(method, this.uri, this, response);
       }
+
+      if (responseInterceptor != null) {
+        response = await responseInterceptor(finalizedRequest, response);
+      }
     } catch (e) {
       var error = e;
       if (!_done.isCompleted) {
         _done.complete();
       }
       cleanUp();
-      checkForCancellation(response: response);
       if (error is! RequestException) {
         error = new RequestException(method, this.uri, this, response, error);
       }
+      if (responseInterceptor != null) {
+        response = await responseInterceptor(finalizedRequest, response, error);
```

Comment:
Good catch (pun intended), there should probably be a try/catch around line 451 that would record that the error happened and then rethrow it.. then we can avoid calling it again and still wrap the error in a `RequestException`